samtranslator/internal/schema_source/aws_serverless_statemachine.py

```diff
@@ -72,6 +72,7 @@ class ScheduleV2EventProperties(BaseModel):
     ScheduleExpressionTimezone: Optional[PassThroughProp] = scheduleeventv2properties("ScheduleExpressionTimezone")
     StartDate: Optional[PassThroughProp] = scheduleeventv2properties("StartDate")
     State: Optional[PassThroughProp] = scheduleeventv2properties("State")
+    OmitName: Optional[bool]  # TODO: add doc
```

Comment:
What would the docs look like?

Reply:
> By default, the `Name` is always set for `AWS::Scheduler::Schedule` for backward compatibility. Set this property to true to omit the `Name`.

Reply:
It doesn't _really_ omit the name from a customer perspective, as it'll still have a name, just whatever CloudFormation decides (instead of SAM). We can probably figure out some better wording for it, so I'm fine with this although it's a little confusing.

Reply:
true, "omit the name in SAM transform"